tests/components/iron_os/test_update.py

```diff
@@ -75,3 +76,32 @@ async def test_update_unavailable(
     state = hass.states.get("update.pinecil_firmware")
     assert state is not None
     assert state.state == STATE_UNAVAILABLE
+
+
+@pytest.mark.usefixtures("ble_device")
+async def test_update_restore_last_state(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    mock_pynecil: AsyncMock,
+) -> None:
+    """Test update entity restore last state."""
+
+    mock_pynecil.get_device_info.side_effect = CommunicationError
+    with patch(
+        "homeassistant.components.iron_os.update.IronOSUpdate.async_get_last_state",
+        return_value=State(
+            "update.pinecil_firmware",
+            STATE_ON,
+            attributes={ATTR_INSTALLED_VERSION: "v2.21"},
+        ),
+    ):
```

Comment:
Instead of mocking async_get_last_state, for restore tests we usually just manually store the previous state and have HA figure it out

Reply:
I believe DSMR does that

Reply:
DSMR didn't do that but i found out there is a helper method for this ^^